codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/MapConstraintViolationGenerator.kt

```diff
@@ -84,8 +88,22 @@ class MapConstraintViolationGenerator(
                     ${if (keyConstraintViolationExists) "##[doc(hidden)] Key(#{KeyConstraintViolationSymbol})," else ""}
                     ${if (valueConstraintViolationExists) "##[doc(hidden)] Value(#{KeySymbol}, #{ValueConstraintViolationSymbol})," else ""}
                 }
+                
+                impl #{Display} for $constraintViolationName {
+                    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+                        return match self {
+                            ${if (shape.hasTrait<LengthTrait>()) "#{LengthMatchingArm}" else ""}
+                            ${if ((keyConstraintViolationExists) || (valueConstraintViolationExists)) """other => write!(f, "{}", other),""" else ""}
```

Comment:
Does this infinitely recurse? Basically self::display -> write -> self::display?

Reply:
Oh good catch! Yes, it will recurse indefinitely. Hmm.. the tests should have caught this. Let me fix the tests as well. Thanks